database/database_test.go

```diff
@@ -234,3 +234,180 @@ func TestStatesWithStatus(t *testing.T) {
 		}
 	}
 }
+
+// setTests is a struct that is used in testing SetIneligibleDomainStates
+var setTests = []struct {
+	description       string
+	domainStates      []IneligibleDomainState
+	wantStatusReports []string
+}{
+	{
+		"two domains",
+		[]IneligibleDomainState{
+			{Name: "youtube.test", Policy: "bulk-1-year"},
+			{Name: "garron.test", Policy: "bulk-1-year"},
+		},
+		[]string{"Updating 2 entries...", " done.\n"},
+	},
+	{
+		"bulk-18-week",
+		[]IneligibleDomainState{{Name: "gmail.test", Policy: "bulk-18-week"}},
+		[]string{"Updating 1 entries...", " done.\n"},
+	},
+	{
+		"bulk-1-year",
+		[]IneligibleDomainState{{Name: "wikipedia.test", Policy: "bulk-1-year"}},
+		[]string{"Updating 1 entries...", " done.\n"},
+	},
+}
+
+// Test SetIneligibleDomainStates
+func TestSetIneligibleDomainStates(t *testing.T) {
+	resetDB()
+
+	for _, tt := range setTests {
+
+		var statuses []string
+		statusReport := func(format string, args ...interface{}) {
+			formatted := fmt.Sprintf(format, args...)
+			statuses = append(statuses, formatted)
+		}
+
+		err := testDB.SetIneligibleDomainStates(
+			tt.domainStates,
+			statusReport,
+		)
+		if err != nil {
+			t.Errorf("[%s] cannot put states %s", tt.description, err)
+			return
+		}
+
+		if !reflect.DeepEqual(statuses, tt.wantStatusReports) {
+			t.Errorf("[%s] Incorrect status reports: %#v", tt.description, statuses)
+		}
+	}
+}
+
+// getAndDeleteTests is a struct that is used in testing GetIneligibleDomainStates
+// and DeleteIneligibleDomainStates
+var getAndDeleteTests = []struct {
+	description  string
+	domainStates []string
+	wantStates   []IneligibleDomainState
+}{
+	{
+		"one domain",
+		[]string{"a.test"},
+		[]IneligibleDomainState{
+			{Name: "a.test", Policy: "bulk-1-year"},
+		},
+	},
+	{
+		"two domains",
+		[]string{"b.test", "c.test"},
+		[]IneligibleDomainState{{Name: "b.test", Policy: "bulk-18-week"},
+			{Name: "c.test", Policy: "bulk-18-week"}},
+	},
+	{
+		"multiple domains",
+		[]string{"e.test", "f.test", "g.test", "h.test", "i.test", "j.test", "k.test"},
+		[]IneligibleDomainState{
+			{Name: "e.test", Policy: "bulk-1-year"},
+			{Name: "f.test", Policy: "bulk-1-year"},
+			{Name: "g.test", Policy: "bulk-18-week"},
+			{Name: "h.test", Policy: "bulk-18-week"},
+			{Name: "i.test", Policy: "bulk-1-year"},
+			{Name: "j.test", Policy: "bulk-18-week"},
+			{Name: "k.test", Policy: "bulk-1-year"},
+		},
+	},
+}
+
+// Test GetIneligibleDomainStates
+func TestGetIneligibleDomainStates(t *testing.T) {
+
+	resetDB()
+
+	// domainStates should be empty as domains are not added to database
+	domainStates, err := testDB.GetIneligibleDomainStates([]string{"a.test"})
+	if len(domainStates) != 0 {
+		t.Errorf("Empty database should contain no preloaded domains")
+	}
```

Comment:
If the intention here is to check that the database is empty, it's fine to use `testDB.backend` directly to get all IneligibleDomainStates (in a helper function that can be used here and in TestDeleteIneligibleDomainStates). It's also fine to assume that `resetDB` has done just that and the database is empty.